src/mpid/ch4/src/ch4_progress.c

```diff
@@ -159,6 +159,26 @@ int MPIDI_Progress_test(int flags)
     return progress_test(&state, 0);
 }
 
+/* provide an internal direct progress function. This is used in e.g. RMA, where
+ * we need poke internal progress from inside a per-vci lock.
+ */
+int MPIDI_progress_test_vci(int vci)
```

Comment:
How can we be sure progressing a single VCI will be sufficient to complete a window epoch? I suspect that could result in deadlock for some scenarios. E.g. if send/recv are on different VCIs than window sync functions, a recv may block because the sender locally buffered the send data.
```
Process 0                        Process 1
MPI_Send (locally buffered)      MPI_Recv
MPI_Win_fence(win)               MPI_Win_fence(win)
```

Reply:
I see this gets fixed up later in the PR. Perhaps it can be merged into this commit, and the commit message amended. Right now it states that global progress is unnecessary.

Reply:
Merged.